.changeset/little-turkeys-hammer.md

```diff
@@ -0,0 +1,8 @@
+---
+'@finos/legend-graph': patch
+'@finos/legend-studio': patch
+---
+
+Due to discrepancies in the test runners for mapping and service, we don't need to do any (un)escaping for the expectedResult assertion data of the service test during its initialization and generation like what we do for mapping test assertion data. So, removing the toGrammarString & fromGrammarString pair for service test runner to avoid it adding additional escape characters. For better context:
```

Comment:
```suggestion
Fix a problem with escaping of single quote character which causes service tests created in Studio fail ([#586](See https://github.com/finos/legend-studio/issues/586)), this can be considered a workaround until we figure out a strategy for the discrepancies in mapping test and service test runners in `Engine` (see [finos/legend-engine#429](https://github.com/finos/legend-engine/issues/429))
```